app/controllers/staff/activity_forms_controller.rb

```diff
@@ -87,6 +87,7 @@ def update
     end
 
     update_form_state
+    add_transparency_identifier
```

Comment:
As a result of the recent refactor here I think you can move this logic to the identifier step https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/541/commits/a4acb9a64d5788984e0ce489a0e44f218a33d481#diff-825521bf19c854ba561043b5a2d09f21R59 and potentially remove the need for guards.